### PR TITLE
feat: add Simple Roster Webhook to Remote environment

### PR DESCRIPTION
**Pull request by SergiiTao**
Related to: https://oat-sa.atlassian.net/browse/LSI-1379

it is **DRAFT** because it has not merged yet dependency

depends on https://github.com/oat-sa/tao-core/pull/3766

How to test:
1. leave the webhook field empty. notice that no webhook was triggered after the successful publishing
2. create multiple webhooks triggered by `Remote Delivery Created Event`, select one in the `Remote environment` config, see that only the chosen one was triggered after the successful publishing

image

### CODE REVIEW

File: test/unit/model/publishing/delivery/tasks/RemoteTaskStatusSynchroniserTest.php [hunk 1]
```diff
@@ -188,7 +191,8 @@ public function testInvoke_WhenRemoteTaskNotFinished_ReturnsInfoReport()
             'DUMMY_REMOTE_TASK_ID',
             'DUMMY_ENVIRONMENT_ID',
             'DUMMY_DELIVERY_ID',
-            'DUMMY_TEST_ID'
+            'DUMMY_TEST_ID',
+            'DUMMY_WEBHOOK_ID'
```
Comment (taorafael) [inline test/unit/model/publishing/delivery/tasks/RemoteTaskStatusSynchroniserTest.php:195]:
```suggestion
            'DUMMY_WEBHOOK_ID'
```

File: test/unit/model/publishing/delivery/tasks/RemoteTaskStatusSynchroniserTest.php [hunk 2]
```diff
@@ -167,7 +169,8 @@ public function testInvoke_WhenResponseStatusNotSuccessful_ReturnsErrorReport()
...
```
Comment (taorafael) [inline test/unit/model/publishing/delivery/tasks/RemoteTaskStatusSynchroniserTest.php:173]:
```suggestion
            'DUMMY_WEBHOOK_ID',
```

File: test/unit/model/publishing/delivery/tasks/RemoteTaskStatusSynchroniserTest.php [hunk 3]
```diff
@@ -128,7 +129,8 @@ public function testInvoke_WhenRemoteTaskFailed_ReturnsErrorReport()
...
```
Comment (taorafael) [inline test/unit/model/publishing/delivery/tasks/RemoteTaskStatusSynchroniserTest.php:133]:
```suggestion
            'DUMMY_WEBHOOK_ID',
```

File: test/unit/model/publishing/delivery/tasks/RemoteTaskStatusSynchroniserTest.php [hunk 4]
```diff
@@ -97,7 +97,8 @@ public function testInvoke_WhenRemoteTaskCompleted_ReturnsSuccessReport()
...
```
Comment (taorafael) [inline test/unit/model/publishing/delivery/tasks/RemoteTaskStatusSynchroniserTest.php:101]:
```suggestion
            'DUMMY_WEBHOOK_ID',
```

**Review by taorafael — APPROVED** [review]
Code-wise it seems fine

File: model/publishing/delivery/tasks/RemoteTaskStatusSynchroniser.php [hunk 5]
```diff
@@ -52,10 +52,10 @@ public function getRemoteStatus()
      */
     public function __invoke($params)
     {
-        if (count($params) != 4) {
+        if (count($params) !== 5) {
```
[the remark this answers is not in the document]
Comment (SergiiTao) [reply model/publishing/delivery/tasks/RemoteTaskStatusSynchroniser.php:55]: you're right. old tasks will fail

File: model/publishing/event/RemoteDeliveryCreatedEvent.php [hunk 6]
```diff
@@ -41,12 +43,21 @@ class RemoteDeliveryCreatedEvent extends AbstractDeliveryEvent implements Webhoo
     /** @var string|null */
     private $alias;
 
-    public function __construct(string $deliveryUri, string $testUri, string $remoteDeliveryUri, string $alias = null)
-    {
+    /** @var string|null */
+    private $webhookId;
```
Comment (andreluizmachado) [inline model/publishing/event/RemoteDeliveryCreatedEvent.php:47]:
```suggestion
    private ?string $webhookId;
```

File: model/publishing/delivery/tasks/RemoteDeliveryPublishingTask.php [hunk 7]
```diff
@@ -56,6 +58,9 @@ class RemoteDeliveryPublishingTask implements Action, ServiceLocatorAwareInterfa
     /** @var core_kernel_classes_Resource */
     private $environment;
 
+    /** @var string|null */
+    private $webhookId;
```
Comment (andreluizmachado) [inline model/publishing/delivery/tasks/RemoteDeliveryPublishingTask.php:62]:
```suggestion
    private ?string $webhookId;
```

File: model/publishing/event/RemoteDeliveryCreatedEvent.php (hunk 6 above)
Comment (SergiiTao) [reply model/publishing/event/RemoteDeliveryCreatedEvent.php:47]: I expect it to be backported to PHP7.4 versions

**Review by mianchenia — APPROVED** [review]

**Review by uncleempty — COMMENTED** [review]
In general - looks good and I would approve it, but IMO there is an interesting point to discuss, so this would be a comment only:
I do agree with all the changes except naming of fields/rdf properties/etc. Talking only about those having `...SimpleRoster...` in name. IMO - TAO platform doesn't know if it's Simple Roster, MorePowerful Roster, or WhateverElse Roster. It doesn't and it shouldn't.
Suggestion:
rename all mentions of SR to be just a `...publishing...`, like `PROPERTY_SIMPLE_ROSTER_WEBHOOK` would change to `PROPERTY_PUBLISHING_WEBHOOK`.
Reason:
This can be used not only with SR, but like a generic webhook fired after publishing a delivery to remote.
Optional:
Make input and it's label visible on the form, only if `Remote Publishing` checkbox is ticked

**Pull request merged**